src/DeviceInput/deviceInputSystem.ts

```diff
@@ -103,6 +103,19 @@ export class DeviceInputSystem implements IDisposable {
         return device[inputIndex];
     }
 
+    /**
+     * Checks for existing connections to devices and register them, if necessary
+     */
+    public checkForConnectedDevices() {
+        const gamepads = navigator.getGamepads();
+
+        for (let i = 0; i < gamepads.length; i++) {
+            if (gamepads[i]) {
+                this._addGamePad(gamepads[i]);
+            }
+        }

```

Comment:
nit: this avoids indexing into the array twice
```suggestion
        for (const gamepad of gamepads) {
            if (gamepad) {
                this._addGamePad(gamepad);
            }
        }
```